assayist/processor/container_rpm_analyzer.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from assayist.processor.base import Analyzer
+from assayist.processor.utils import get_koji_session
+from assayist.processor.logging import log
+from assayist.common.models import content
+
+
+class ContainerRPMAnalyzer(Analyzer):
+    """Analyze the RPMs in a container image layer."""
+
+    def run(self):
+        """Start the container RPM analyzer."""
+        build_info = self.read_metadata_file(self.BUILD_FILE)
+        build_id = build_info['id']
+        if not self.is_container_build(build_info):
+            log.info(f'Skipping build {build_id} because the build is not a container')
+            return
+
+        self.koji_session = get_koji_session()
+        # Create a mapping of arch to archive (container image) so we can easily map to the
+        # parent container archives in a future loop
+        arch_to_archive = {}
+        not_container_msg = 'Skipping archive {0} since it\'s not a container image'
+        for archive in self.koji_session.listArchives(build_id):
+            if not archive['filename'].startswith('docker-image'):
+                log.debug(not_container_msg.format(archive['id']))
+                continue
+            arch = archive['extra']['image']['arch']
+            if arch in arch_to_archive:
+                log.error(
+                    f'Build {build_id} has more than one container image with the arch {arch}')
+                continue
+            arch_to_archive[arch] = archive
+
+        parent_build_id = build_info['extra']['image'].get('parent_build_id')
+        # If there is a parent to this image, then only get the RPMs installed in this layer
+        # and mark them as embedded artifacts on this container image
+        if parent_build_id is not None:
+            # Find the RPMs installed in this layer versus the parent image
+            for archive in self.koji_session.listArchives(parent_build_id):
+                if not archive['filename'].startswith('docker-image'):
+                    log.debug(not_container_msg.format(archive['id']))
+                    continue
+                arch = archive['extra']['image']['arch']
+                if arch not in arch_to_archive:
+                    log.debug(
+                        f'The parent build {parent_build_id} contains an extra arch of {arch}')
+                    continue
+
+                rpms = self._get_rpms_diff(archive['id'], arch_to_archive[arch]['id'])
+                self._process_embedded_rpms(archive, rpms)
```

Comment:
I think we're passing the parent archive (`archive`) to _process_embedded_rpms instead of the child (`arch_to_archive[arch]`).

Reply:
Good catch!